core/templates/components/skill-selector/skill-selector.component.ts

```diff
@@ -0,0 +1,213 @@
+// Copyright 2018 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Controller for the select skill viewer.
+ */
+
+import { Component, OnInit, Input, Output, EventEmitter } from '@angular/core';
+import { CategorizedSkills } from 'domain/topics_and_skills_dashboard/topics-and-skills-dashboard-backend-api.service';
+import { SkillSummary } from 'core/templates/domain/skill/skill-summary.model';
+import { downgradeComponent } from '@angular/upgrade/static';
+import { GroupedSkillSummaries } from 'pages/skill-editor-page/services/skill-editor-state.service';
+import { FilterForMatchingSubstringPipe } from 'filters/string-utility-filters/filter-for-matching-substring.pipe';
+import { ShortSkillSummary } from 'core/templates/domain/skill/ShortSkillSummaryObjectFactory';
+
+require('domain/utilities/url-interpolation.service.ts');
```

Comment:
I don't think this is required

Reply:
Use angular DI

Reply:
Could you clarify how to use DI here?

Reply:
This is not required. I don't see the use of UrlInterpolationService in this file anymore. Incase it was used, the way to do it would be:
```
import { UrlInterpolationService } from 'domain/utilities/url-interpolation.service.ts';
...
constructor( private urlInterpolationSerivce: UrlInterpolationService){}
...
```

Reply:
https://angular.io/guide/dependency-injection